### DISCUSSION TITLE
Feature Request: Notify User when we load settings from environment variables

### DISCUSSION THREAD

**Discussion by Mause** [duckdb/duckdb#7775] [Ideas]
With the addition of https://github.com/duckdb/duckdb/pull/5419, a number of users have reported their workflows breaking in 0.7.x (https://github.com/duckdb/duckdb/issues/6563, https://github.com/duckdb/duckdb/issues/6786). This is due to settings being loaded, either completely or partially (some keys manually set, some from the environment), and causing auth failures. Having this happen silently is not ideal, but presently DuckDB has no standardised way of notifying users of this situation. 

It's possible this could be solved by switching to the proper AWS credential chain instead of cherry-picking environment variables

**Comment by mskyttner** [comment]
For reading from S3-compatible storage with connection info from environment variables, minio allows hybrid-cloud setups / self-hosting and is often used as an open source alternative to AWS S3. Since 2015 the aws cli doesn't have the ability to specify an endpoint url in a profile (see https://stackoverflow.com/questions/52494196/is-there-any-way-to-specify-endpoint-url-in-aws-cli-config-file) so if switching to the proper AWS credential chain involves depending on the aws cli and such profiles it appears users of S3 storage backed by minio would face some extra quirks to deal with?

**Reply by Mause** [reply]
We're definitely not removing support for other clouds, don't worry - we actually use minio to run our httpfs integration tests. I'm mostly using S3 here to refer to the "protocol"
